src/hotspot/share/opto/loopnode.cpp

```diff
@@ -1106,8 +1106,8 @@ int PhaseIdealLoop::extract_long_range_checks(const IdealLoopTree* loop, jlong s
         jlong scale = 0;
         if (loop->is_range_check_if(if_proj, this, T_LONG, phi, range, offset, scale) &&
             loop->is_invariant(range) && loop->is_invariant(offset) &&
-            original_iters_limit / ABS(scale * stride_con) >= min_iters) {
-          reduced_iters_limit = MIN2(reduced_iters_limit, original_iters_limit/ABS(scale));
```

Comment:
Mixing signed and unsigned can cause compiler warnings and a change in semantics.  And again, checked_cast is not better than a raw cast here.  I think it would be better to make min_iters, reduced_iters_limit, original_iters_limit all unsigned.